mcs/class/System.Drawing/System.Drawing/Graphics.cs

```diff
@@ -288,6 +288,9 @@ public void Dispose ()
 		{
 			Status status;
 			if (! disposed) {
+				if (deviceContextHdc != IntPtr.Zero)
```

Comment:
I think it would be better to do this check inside of `ReleaseHdc ()`

Reply:
We already have a test showing that calling ReleaseHdc before GetHdc throws an exception.